pkg/provisioning/provisioning.go

```diff
@@ -0,0 +1,86 @@
+package provisioning
+
+import (
+	"github.com/metalkube/baremetal-operator/pkg/bmc"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("provisioner")
+
+/*
+Package provisioning defines the API for talking to the provisioning backend.
+*/
+
+// Provisioner holds the state information for talking to the
+// provisioning backend.
+//
+// NOTE(dhellmann): Provisioner will eventually become an interface,
```

Comment:
Using an interface makes testing a lot easier, so I would suggest doing that now. It's a common approach for enabling unit tests of hard API boundaries like this.